### DISCUSSION TITLE
Is it possible to dynamically change the --output PATH ?

### DISCUSSION THREAD

**Discussion by jvthomas90** [lncrawl/lightnovel-crawler#2190] [Q&A]
The default behavior _(on my Mac, the location will obviously differ for others who may use another platform)_ is to create a "Lightnovels" folder in my home directory, and then create subfolders based on source site, then the novel name, and finally the download contents are contained in that 3rd layer.

I usually manually delete the source/site folder, but I recently got tired of the repetetive manual backspacing and decided to test out the `-o` flag. Now everything redirects to ~/Lightnovels/ _(or whatever other folder I may choose by the end of this experimental period)_ but I noticed that subfolders are no longer being created. 

This isn't really a problem if you're only ever planning on having one single novel downloaded to that folder at any given time, but as soon as you introduce 2 or more downloads... all of the `meta.json` and `cover.jpg` files and even the `format` folders will either get duplicated, renamed, overwritten, etc _(not sure of the exact behavior as I didn't take my test that far, I just saw that an overwrite was about to happen and stopped my test preemptively. Suffice it to say, one way or another, it'll become a disorganized mess)_

### **TL;DR** is there some way to combine the value for the `--output` flag in conjunction with a dynamically evaluated variable as the endpoint?  i.e. `lncrawl -o ~/calibre/imports/$NOVEL --filename-only` this way, just like how the default behavior auto-creates a subfolder with the novel name, the `-o` flag doesn't need to stay static either. 

**P.S.** I'm honestly not too tech savvy, though I do try my best to learn. Nevertheless, if my ignorance shows in any above assumptions, I apologize. I'm honestly just curious if this is even possible! 🤞🏽

**Comment by alzamer2** [comment]
why not create proxy python file?
here is simple proxy python for lncrawl 
[lncrawl_proxy.zip](https://github.com/dipu-bd/lightnovel-crawler/files/13741799/lncrawl_proxy.zip)
this script will import all arg to lncrawl and i added new arg with is -o_n ( or --output_novel)
this arg will extract name source url and add it to output path
just extract it and use lncrawl_proxy.py as you use lncrawl 
example
`lncrawl_proxy.py -s https://www.wuxiaworld.com/novel/nine-star-hegemon -o_n ~/calibre/imports/`
